app/models/study.rb

```diff
@@ -1132,6 +1132,22 @@ def unique_genes
     Gene.where(study_id: self.id, :study_file_id.in => self.expression_matrix_files.map(&:id)).pluck(:name).uniq
   end
 
+  # Get species scientific name for a searched gene.  E.g. PTEN -> Homo sapiens.
+  def infer_species(searched_gene)
+    # Get gene record for searched gene string
+    gene = self.genes.by_name_or_id(searched_gene, self.expression_matrix_files.map(&:id))
+
+    # Get matrices containing this gene.  (Same gene name can be in multiple matrices.)
+    study_file_ids = Gene.where(study_id: self.id, name: gene['name']).pluck(:study_file_id)
+
+    # Get species for each matrix.
+    # TODO (): Handle when a searched gene maps to multiple species
+    taxon_ids = StudyFile.where(:_id.in => study_file_ids).pluck(:taxon_id)
+    scientific_name = Taxon.where(:_id.in => taxon_ids).pluck(:scientific_name)[0]
```

Comment:
```suggestion
    scientific_name = Taxon.where(:_id.in => taxon_ids).pluck(:scientific_name).first
```

Reply:
Incorporated as part of other refinements; see [here](https://github.com/broadinstitute/single_cell_portal_core/blob/1731b32e9d0f2a7fbd6d89c177b5eebc95ca0641/app/models/study.rb#L1135).